### ISSUE TITLE
When content/index.md exists, no other files in content get rendered. 

### ISSUE THREAD

**Issue by looeee** [gohugoio/hugo#4550]
Steps to reproduce:

Create new empty site. 

Create a page in content and corresponding layout, e.g.

`/content/test.md`

```
---
title: "Test"
date: 2018-04-01T09:00:44+07:00
type: page
layout: test
---

JUST TESTING!
```

`/layout/page/test.html`

```
{{ .Content }}
```

Now the page will render correctly. However, add `/content/index.md`, which can contain anything and have any layout (as far as I can tell), and suddenly `test.md` and any other content in `/content` stop being rendered. 

I've tested this with and without `/layout/index.html` existing, it doesn't seem to change anything.

**Comment by bep** [comment]
Please use https://discourse.gohugo.io/ for questions/troubleshooting. Also see  [Hugo Documentation](http://gohugo.io/overview/introduction/).

**Comment by looeee** [comment]
OK, I tried raising this on [discourse](https://discourse.gohugo.io/t/when-content-index-md-exists-no-other-files-in-content-get-rendered/11328/4) as well and it just got shut down straight away. 

I've re-read the documentation page about "leaf bundles", so at least I understand what was happening now. But I don't think that the documentation is clear enough - I have read that page before, but it did not lead me to expect that the index.md at the root of the content folder would be treated the same way as, for example, `content/blog/index.md`. 

However, I've tried to bring this up in both forums now and been shut down without discussion straight away, so clearly you don't want to engage with users who have feedback like this.
